Mathlib/Lean/Expr/Basic.lean

```diff
@@ -194,6 +194,19 @@ def bvarIdx? : Expr → Option Nat
 def getAppFnArgs (e : Expr) : Name × Array Expr :=
   withApp e λ e a => (e.constName, a)
 
+/-- Invariant: `i : ℕ` should be less than the size of `as : Array Expr`. -/
+private def getAppAppsAux : Expr → Array Expr → Nat → Array Expr
+  | .app f a, as, i => getAppAppsAux f (as.set! i (.app f a)) (i-1)
+  | _,       as, _ => as
+
+/-- Given `f a b c`, return `#[f a, f a b, f a b c]`.
+    Remark: this procedure is very similar to `getAppArgs`. -/
```

Comment:
What is this for? It seems an unlikely thing to want in general?

Reply:
This is used in the `cc` tactic.